### DISCUSSION TITLE
MySQLContainer unable to start when using withConfigurationOverride()

### DISCUSSION THREAD

**Discussion by billclar99** [testcontainers/testcontainers-java#6304] [Q&A] [answered]
TestContainers Version:  1.17.6v2.13.0
Docker Compose:  v2.13.0

I'm running a MySQLContainer for 8.0.31 and I'm specifying a specific configuration override file:

`container = (new TestMySQLContainer())`
`		.withUsername("root")`
`		.withPassword("")`
`		.withDatabaseName("temp_database")`
`		.withConfigurationOverride("mysql-configuration-override");`

mysql-configuration-override contains:

[mysqld]
sql_mode=NO_ENGINE_SUBSTITUTION
default-time-zone='US/Eastern'

The container fails to start.  This is the output from the mysql container logs:

 2022-12-14 16:28:41+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.31-1.el8 started.
 2022-12-14 16:28:41+00:00 [ERROR] [Entrypoint]: mysqld failed while attempting to check config
 	command was: mysqld --verbose --help --log-bin-index=/tmp/tmp.6Beq3pooyu
 	mysqld: Can't read dir of '/etc/mysql/conf.d/' (OS errno 20 - Not a directory)
 mysqld: [ERROR] Stopped processing the 'includedir' directive in file /etc/my.cnf at line 36.
 mysqld: [ERROR] Fatal error in defaults handling. Program aborted!

I looked at the mysql image and this is the only step I could find that references /etc/mysql/conf.d:

set -eux; 
microdnf install -y "mysql-community-server-minimal-$MYSQL_VERSION"; 
microdnf clean all; 
grep -F 'socket=/var/lib/mysql/mysql.sock' /etc/my.cnf;
sed -i 's!^socket=.*!socket=/var/run/mysqld/mysqld.sock!' /etc/my.cnf; 
grep -F 'socket=/var/run/mysqld/mysqld.sock' /etc/my.cnf; 
{ echo '[client]'; echo 'socket=/var/run/mysqld/mysqld.sock'; } >> /etc/my.cnf; 
! grep -F '!includedir' /etc/my.cnf; 
{ echo; echo '!includedir /etc/mysql/conf.d/'; } >> /etc/my.cnf; 
mkdir -p /etc/mysql/conf.d; 
mkdir -p /var/lib/mysql /var/run/mysqld; 
chown mysql:mysql /var/lib/mysql /var/run/mysqld; 
chmod 1777 /var/lib/mysql /var/run/mysqld; 
mkdir /docker-entrypoint-initdb.d; 
mysqld --version; 
mysql --version

**Comment by eddumelendez** [comment, marked as answer]
`mysql-configuration-override` is the folder where the conf file should be. So, having `src/test/resources/mysql-configuration-override/my.cnf` created will make it work. Look at [this example](https://github.com/testcontainers/testcontainers-java/blob/f5e79b9aab308acc0211fca4bd0d00ab9b57cbd6/modules/mysql/src/test/java/org/testcontainers/junit/mysql/SimpleMySQLTest.java#L80-L92)

I did a similar test with mysql:8.0.31 and works fine.

**Reply by billclar99** [reply]
That fixed it.  Thank you!

**Comment by cklein-arcadia** [comment]
Ran into the same thing trying to configure ScyllaDB's testcontainer. In the end the same approach solved the problem for me:
```
ScyllaDBContainer scylladb = new ScyllaDBContainer(DockerImageName.parse("scylladb/scylla:6.2"))
        .withConfigurationOverride("scylla");
```

And then I created a file at `src/test/resources/scylla/scylla.yaml` with these contents:
```
replication_strategy_warn_list: []
```

Just recording this for future self that has to reconfigure the Scylla testcontainer!
